pkg/pillar/cmd/zedmanager/zedmanager.go

```diff
@@ -742,3 +772,76 @@ func handleGlobalConfigDelete(ctxArg interface{}, key string,
 	*ctx.globalConfig = *types.DefaultConfigItemValueMap()
 	log.Functionf("handleGlobalConfigDelete done for %s", key)
 }
+
+func handleZedAgentStatusCreate(ctxArg interface{}, key string,
+	statusArg interface{}) {
+	handleZedAgentStatusImpl(ctxArg, key, statusArg)
+}
+
+func handleZedAgentStatusModify(ctxArg interface{}, key string,
+	statusArg interface{}, oldStatusArg interface{}) {
+	handleZedAgentStatusImpl(ctxArg, key, statusArg)
+}
+
+func handleZedAgentStatusImpl(ctxArg interface{}, key string,
+	statusArg interface{}) {
+	ctxPtr := ctxArg.(*zedmanagerContext)
+	status := statusArg.(types.ZedAgentStatus)
+	if status.RebootCmd {
+		filterAppInstancesWithLocalProfile(ctxPtr, true)
+	} else if ctxPtr.currentProfile != status.CurrentProfile {
+		ctxPtr.currentProfile = status.CurrentProfile
+		filterAppInstancesWithLocalProfile(ctxPtr, false)
+	}
+	log.Functionf("handleZedAgentStatusImpl(%s) done", key)
+}
+
+// filterAppInstancesWithLocalProfile check all app instances with currentProfile and set activate state
+func filterAppInstancesWithLocalProfile(ctx *zedmanagerContext, shutdown bool) {
```

Comment:
This does more than filltering since it calls doUpdate.

Maybe the name should be updateBasedOnProfile?

Reply:
Sounds reasonable, corrected.